### ISSUE TITLE
Error in class Dist if self.base_path != self.path

### ISSUE THREAD

**Issue by cyrilbouvier** [waf-project/waf#1806]
Hi,

There is a problem in the method `archive` of the class `Dist` (in `Scripting.py`).
If one define self.base_path to be different to self.path, the script ends with an error.
Indeed, the node containing the future archive is computed as
```
node=self.base_path.make_node(arch_name)
```

but in the following code the zip and tar archive are created as follow

```
tar=tarfile.open(arch_name,'w:'+self.algo.replace('tar.',''))
....
zip=zipfile.ZipFile(arch_name,'w',compression=zipfile.ZIP_DEFLATED)
```

Basically the self.base_path component of the node is forgotten leading to errors in the rest of the code.

Best regards,
Cyril

**Comment by ita1024** [comment]
Update your copy to f7367c805128c931e40475791599f853c7cbfb48

**Comment by cyrilbouvier** [comment]
The problem is fixed with commit f7367c8
